src/web/cursorUtils.ts

```diff
@@ -57,31 +64,29 @@ function moveCursorToEnd(target: HTMLElement) {
   }
 }
 
-function getIndexedPosition(target: HTMLElement, range: Range, isStart: boolean) {
-  const marker = document.createTextNode('\0');
-  const rangeClone = range.cloneRange();
+function getCurrentCursorPosition(target: HTMLElement) {
+  const sel = window.getSelection();
 
-  rangeClone.collapse(isStart);
+  if (sel && sel.rangeCount > 0) {
+    const range = sel.getRangeAt(0);
+    const preSelectionRange = range.cloneRange();
+    preSelectionRange.selectNodeContents(target);
+    preSelectionRange.setEnd(range.startContainer, range.startOffset);
+    const start = preSelectionRange.toString().length;
 
-  rangeClone.insertNode(marker);
-  const position = target.innerText.indexOf('\0');
-  if (marker.parentNode) {
-    marker.parentNode.removeChild(marker);
+    return {
+      start,
+      end: start + range.toString().length,
+    };
   }
-
-  return position;
+  return {start: -1, end: -1};
```

Comment:
We return -1 in case selection wasn't found so we can detect and handle this situation in other pars of the code. I will change in `getCurrentCursorPosition` in `parserUtils` so when cursorPosition is -1, we will move cursor to the end. What do you think about it?

Reply:
-1 is a hacky value and it can go undetected and cause some troubles.

Can we return `null` instead of `{start: -1, end: -1}` and add null check in other parts of the code you've mentioned?